testingutils/src/test/java/com/github/awsjavakit/testingutils/networking/WiremockDirectCallClientTest.java

```diff
@@ -0,0 +1,263 @@
+    var directCallHttpServer = factory.getHttpServer();
+    directCallClient = new WiremockDirectCallClient(directCallHttpServer);
+  }
+
+  @AfterEach
+  public void stop() {
+    this.directCallServer.stop();
+  }
+
+  @ParameterizedTest
+  @MethodSource("requestProvider")
+  void shouldTransformWiremockResponseForRequests(TestSetup testSetup)
+    throws IOException, InterruptedException {
+    var expectedResponseBody = testSetup.expectedResponse().getBody();
+    var expectedResponseCode = testSetup.expectedResponse().getStatus();
+    var expectedHeaders = new ConcurrentHashMap<>(
+      HeadersUtils.wiremockHeadersToJavaHeaders(testSetup.expectedResponse().getHeaders()).map());
+    var mapping = testSetup.mapping();
+
+    directCallServer.stubFor(mapping);
+
+    var request = testSetup.request();
+    var response = directCallClient.send(request, BodyHandlers.ofString());
+
+    assertThat(response.body(), response.statusCode(), is(equalTo(expectedResponseCode)));
+    assertThat(response.body(), is(equalTo(expectedResponseBody)));
+    var responseHeaders = new ConcurrentHashMap<>(response.headers().map());
+    responseHeaders.remove("Matched-Stub-Id");
+    assertThat(responseHeaders, is(equalTo(expectedHeaders)));
+  }
+
+  @Test
+  void shouldThrowExceptionWhenCallingUnsupportedMethod() {
+    assertThrows(UnsupportedOperationException.class, directCallClient::cookieHandler);
+    assertThrows(UnsupportedOperationException.class, directCallClient::connectTimeout);
+    assertThrows(UnsupportedOperationException.class, directCallClient::followRedirects);
+    assertThrows(UnsupportedOperationException.class, directCallClient::proxy);
+    assertThrows(UnsupportedOperationException.class, directCallClient::sslContext);
+    assertThrows(UnsupportedOperationException.class, directCallClient::sslParameters);
+    assertThrows(UnsupportedOperationException.class, directCallClient::authenticator);
+    assertThrows(UnsupportedOperationException.class, directCallClient::version);
+    assertThrows(UnsupportedOperationException.class, directCallClient::executor);
+    assertThrows(UnsupportedOperationException.class,
+      () -> directCallClient.sendAsync(null, null, null));
+
+  }
+
+  private static Stream<TestSetup> requestProvider() {
+    return Stream.of(createSetupWithSimpleGetRequest(),
+      createSetupWithRequestWithQueryParams(GET),
+      createSetupWithRequestWithHeaders(GET),
+      createSetupWithRequestWithBody(POST),
+      createSetupWithRequestWithEmptyBody(POST),
+      createSetupWithRequestWithQueryParams(POST),
+      createSetupWithRequestWithHeaders(POST),
+      createSetupWithRequestWithBody(PUT),
+      createSetupWithRequestWithEmptyBody(PUT),
+      createSetupWithRequestWithQueryParams(PUT),
+      createSetupWithRequestWithHeaders(PUT),
+      createSetupWhereResponseHasHeaders(GET),
+      createSetupWhereResponseHasHeaders(POST),
+      createSetupWhereResponseHasHeaders(PUT));
```

Comment:
These are grouped by input, but I might have grouped by method and then extracted and done stream::concat